src/sensors/src/PredictSensorsMeasurements.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2015 Fondazione Istituto Italiano di Tecnologia
+ * Author: Naveen Kuppuswamy
+ * email:  naveen.kuppuswamyt@iit.it
+ * Permission is granted to copy, distribute, and/or modify this program
+ * under the terms of the GNU General Public License, version 2 or any
+ * later version published by the Free Software Foundation.
+ *
+ * A copy of the license can be found at
+ * http://www.robotcub.org/icub/license/gpl.txt
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details
+ */
+
+#include <iDynTree/Sensors/PredictSensorsMeasurements.h>
+#include <iDynTree/Core/SpatialMotionVector.h>
+#include <iDynTree/Sensors/Accelerometer.h>
+#include <iDynTree/Sensors/Gyroscope.h>
+#include <iDynTree/Sensors/Sensors.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/Traversal.h>
+#include <iDynTree/Model/ForwardKinematics.h>
+#include <iDynTree/Model/FreeFloatingState.h>
+#include <iDynTree/Core/SpatialAcc.h>
+#include <map>
+#include <iDynTree/Core/VectorDynSize.h>
+
+
+
+namespace iDynTree {
+PredictSensorsMeasurements::PredictSensorsMeasurements()
+{
+// All local variables, so nothing to do really.
+}
+PredictSensorsMeasurements::~PredictSensorsMeasurements()
+{
+// All local variables, so nothing to do really.
+}
+ 
+bool PredictSensorsMeasurements::operator()(const Model& model,const Traversal& traversal,const iDynTree::FreeFloatingPos& robotPos,const iDynTree::FreeFloatingVel& robotVel,iDynTree::FreeFloatingAcc& robotAcc,const LinAcceleration& gravity,const iDynTree::SensorsList &sensorsList,iDynTree::SensorsMeasurements &predictedMeasurement)
+{
+    
+    bool returnVal = true;
+    // incorporating gravity into the base LinAcceleration
+    iDynTree::LinkPositions linkPos(model);
+    iDynTree::LinkVelArray linkVel(model);
+    iDynTree::LinkAccArray linkAcc(model);
+    
+    
+    iDynTree::AngAcceleration nullAngAccl;
```

Comment:
Can you explicitly set the iDynTree::AngAcceleration to zero? As part of the performance improvement now vectors are not initialized to zero. See https://github.com/robotology/idyntree/issues/98#issuecomment-158999757 .